File: todo-project/src/main/java/kotlinstudy/todoproject/todo/api/model/TodoListResponse.java
package kotlinstudy.todoproject.todo.api.model;

import kotlinstudy.todoproject.todo.domain.Todo;
import lombok.Data;

import java.util.List;
import java.util.stream.Collectors;

@Data
public class TodoListResponse {

    private final List<TodoResponse> items;

    private TodoListResponse(List<TodoResponse> items) {
        this.items = items;
    }

    public int size() {
        return items.size();
    }

    public TodoResponse get(int index) {
        return items.get(index);
    }

    public static TodoListResponse of(List<Todo> todoList) {
        List<TodoResponse> todoListResponse = todoList.stream()
            .map(TodoResponse::of)
            .collect(Collectors.toList());

        return new TodoListResponse(todoListResponse);
    }

}

File: todo-project/src/main/java/kotlinstudy/todoproject/todo/api/model/TodoListResponse.kt
package kotlinstudy.todoproject.todo.api.model

import com.fasterxml.jackson.annotation.JsonIgnore
import kotlinstudy.todoproject.todo.domain.Todo

data class TodoListResponse(
    val items: List<TodoResponse>,
) {

//    fun size() : Int = items.size
    val size: Int
        @JsonIgnore // 다른 곳에서 참조 시 .size 가 안 되도록 설정
        get() = items.size

    fun get(index: Int) = items[index]

    companion object {
        @JvmStatic
        fun of(todoList: List<Todo>) =
             TodoListResponse(todoList.map(TodoResponse::of))
    }
}